File: src/zelpage/razeni/bo/Vehicle.java
package zelpage.razeni.bo;

import java.io.Serializable;
import org.json.JSONObject;

/**
 * Vehicle Business object
 * @author jindrichbr
 */
public class Vehicle implements Serializable{
    private String name;
    private String annotation;

    public String getAnnotation() {
        return annotation;
    }

    public void setAnnotation(String annotation) {
        this.annotation = annotation;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
    
    /**
     * Creates Vehicle object from Json object
     * @param json json object
     * @return new Vehicle object
     */
    public static Vehicle fromJson(JSONObject json){
        Vehicle result = new Vehicle();
        result.setName(json.optString("Name"));
        result.setAnnotation(json.optString("Annotation"));
        return result;
    }
}

File: app/src/main/java/zelpage/razeni/Vehicle.kt
package zelpage.razeni

import org.json.JSONObject
import java.io.Serializable

/**
 * Vehicle Business object
 */
data class Vehicle (
	var name: String? = null,
	var annotation: String? = null) : Serializable {

	companion object {

		/**
		 * Creates Vehicle object from Json object
		 * @param json json object
		 *
		 * @return new Vehicle object
		 */
		fun fromJson(json : JSONObject) : Vehicle =
			Vehicle(
				name = json.optString("Name"),
				annotation = json.optString("Annotation"))

	}

}
